src/Types.fs

```diff
@@ -21,32 +21,31 @@ type CaseStrategy =
 
 type DecoderError = string * ErrorReason
 
-type Decoder<'T> = string -> JsonValue -> Result<'T, DecoderError>
+type Decoder<'T> = string -> JsonElement -> Result<'T, DecoderError>
 
-type Encoder<'T> = 'T -> JsonValue
+type Encoder<'T> = 'T -> JsonNode
```

Comment:
❓ What is the distinction between `JsonElement`/`JsonNode`?

Reply:
So, the `System.Text.Json` library exposes two different DOMs. One is optimized for reading-- that's `JsonDocument`/`JsonElement`. And one is optimized for writing/creation/mutation -- that's `JsonNode`/`JsonValue`.

So, for _decoding_, we want to use the read-optimized/immutable `JsonElement`. And for _encoding_, we want to (have to) use the write-optimized/mutable `JsonNode`/`JsonValue`.